tests/prompts/test_common.py

```diff
@@ -86,7 +87,15 @@ def run(inp):
             == 1000000000000000000000000000001
         )
 
-    execute_with_input_pipe(run)
+    if prompt_toolkit_version < (3, 0, 29):
+        inp = create_pipe_input()
+        try:
+            return asyncio.run(run(inp))
+        finally:
+            inp.close()
+    else:
+        with create_pipe_input() as inp:
+            asyncio.run(run(inp))
```

Comment:
I think you can do something like this:

```suggestion
    pipe_input = create_pipe_input()
    if prompt_toolkit_version < (3, 0, 29):
        pipe_input = pipe_input.create()

    with pipe_input as inp:
        asyncio.run(run(inp))
```